mne/viz/evoked.py

```diff
@@ -311,10 +311,30 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
                         for i in bad_ch_idx:
                             if i in idx:
                                 colors[idx.index(i)] = 'r'
-                    for ch_idx in range(len(D)):
+
+                    # find the channels with the least activity
+                    # to map them in front of the more active ones
+                    if zorder == 'std':
+                        def zorder_(D):
+                            return D.std(axis=1).argsort()
+                    elif zorder == 'unsorted':
+                        def zorder_(D):
+                            return list(range(D.shape[0]))
+                    elif not callable(zorder):
+                        error = ('`zorder` must be a function, "std" '
+                                 'or "unsorted", not {}.')
```

Comment:
>And perhaps, add a test where zorder is a callable. Other than that lgtm.
 
That's the last test.

>Well actually, I wonder if we should get rid of these nested functions and just modify z_ord inside the conditionals since the function gets called on line 330 anyway.

Ah right! So like

```
                    if zorder == 'std':
                        # find the channels with the least activity
                        # to map them in front of the more active ones
                        z_ord = D.std(axis=1).argsort()
                    elif zorder == 'unsorted':
                        z_ord = list(range(D.shape[0]))
                    elif not callable(zorder):
                        error = ('`zorder` must be a function, "std" '
                                 'or "unsorted", not {}.')
                        raise TypeError(error.format(type(zorder)))
                    else:
                        z_ord = zorder(D)
```